projects/user_publication/utils.py

```diff
@@ -26,10 +27,20 @@
 ]
 
 
+def add_to_all_sources(pub, source):
```

Comment:
I'm not a fan of abstractions like this. It would be much simpler and more robust to just set the field and save it rather than using `setattr`.